File: HackatonTomcat/WebContent/CryptUtil.js
function encrypt(text){
	var s = new String();
    for (var i = 0; i < text.length; ++i){
	   s = s.concat(BitArray.shred(text.charCodeAt(i)).pushOff(8).toString());
    } 
	var offusque = new String();
	for(var i in s){
		offusque = offusque.concat(s[i]);
		offusque = offusque.concat(Math.round(Math.random()));
	}
		
    var out = new String();	
	var auxByte = new String();
	for(var i in offusque){
		auxByte = auxByte.concat(offusque[i]);
		if(i%8 == 7){
			var number = (BitArray.fromString(auxByte).toNumber()+ 31) % 10176;
			out = out.concat(String.fromCharCode(number));
			auxByte = new String();
		}
	}
		
    return Base64.encode(out);
}

function decrypt(text){
	
	var textDecoded = Base64.decode(text);
	var offusque = new String();
    for (var i = 0; i < textDecoded.length; ++i){
	   offusque = offusque.concat(BitArray.shred((10176 + textDecoded.charCodeAt(i) - 31) % 10176).pushOff(8).toString());
    } 
	
	var desOffusque = new String();
	for(var i in offusque){
		if(i%2 == 0){
			desOffusque = desOffusque.concat(offusque[i]);
		}
	}	
	var out = new String();	
	var auxByte = new String();
	for(var i in desOffusque){
		auxByte = auxByte.concat(desOffusque[i]);
		if(i%8 == 7){
			out = out.concat(String.fromCharCode(BitArray.fromString(auxByte).toNumber()));
			auxByte = new String();
		}
	}
		
    return out;
	
}

File: HackatonTomcat/src/testcrypt/CryptUtil.java
package testcrypt;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

public class CryptUtil {
	
	public static String encrypt(String text){
		String s = new String();
	    for (int i = 0; i < text.length(); ++i){
		   s = s.concat(BitArray.shred((int)text.charAt(i)).pushOff(8).toString());
	    } 
	    String offusque = new String();
		for(int i = 0; i < s.length(); ++i){
			offusque = offusque.concat(""+s.charAt(i));
			offusque = offusque.concat(""+Math.round(Math.random()));
		}
			
		String out = new String();	
		String auxByte = new String();
		for(int i = 0; i < offusque.length(); ++i){
			auxByte = auxByte.concat(""+offusque.charAt(i));
			if(i%8 == 7){
				long number = (BitArray.fromString(auxByte).toNumber()+ 31) % 10176;
				out = out.concat(((char)number)+"");
				auxByte = new String();
			}
		}		
	    return new String(Base64.getEncoder().encode(out.getBytes(StandardCharsets.UTF_16)), StandardCharsets.UTF_16);
	}

	public static String decrypt(String text){
		
		String textDecoded = new String(Base64.getDecoder().decode(text.getBytes(StandardCharsets.UTF_16)), StandardCharsets.UTF_16);
		String offusque = new String();
	    for (int i = 0; i < textDecoded.length(); ++i){
		   offusque = offusque.concat(BitArray.shred((10176 + (int)textDecoded.charAt(i) - 31) % 10176).pushOff(8).toString());
	    } 
		
	    String desOffusque = new String();
		for(int i = 0; i < offusque.length(); ++i){
			if(i%2 == 0){
				desOffusque = desOffusque.concat(offusque.charAt(i)+"");
			}
		}	
		String out = new String();	
		String auxByte = new String();
		for(int i = 0; i < desOffusque.length(); ++i){
			auxByte = auxByte.concat(desOffusque.charAt(i)+"");
			if(i%8 == 7){
				out = out.concat(""+(char)BitArray.fromString(auxByte).toNumber());
				auxByte = new String();
			}
		}
			
	    return out;
		
	}

}
